File: src/main/java/fr/spoutnik87/musicbot_rest/security/SecurityConfiguration.java
package fr.spoutnik87.musicbot_rest.security;

import lombok.Data;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
@Data
public class SecurityConfiguration {
  @Value("${security.secret}")
  private String secret;
  private final long expirationTime = 864_000_000;
  private final String tokenPrefix = "Bearer ";
  private final String headerString = "Authorization";
  private final String signUpUrl = "/user";
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/security/SecurityConfiguration.kt
package fr.spoutnik87.musicbot_rest.security

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class SecurityConfiguration {
    @Value("\${security.secret}")
    lateinit var secret: String

    val expirationTime: Long = 864000000
    val tokenPrefix = "Bearer "
    val headerString = "Authorization"
    val signUpUrl = "/user"
}